src/parser/bison_parser.y

```diff
@@ -724,18 +724,16 @@ opt_order_type:
 // TODO: TOP and LIMIT can take more than just int literals.
 
 opt_top:
-		TOP int_literal { $$ = new LimitDescription($2->ival, kNoOffset); delete $2; }
+		TOP int_literal { $$ = new LimitDescription($2, Expr::makeNullLiteral()); }
 	|	/* empty */ { $$ = nullptr; }
 	;
 
 opt_limit:
-		LIMIT int_literal { $$ = new LimitDescription($2->ival, kNoOffset); delete $2; }
-	|	LIMIT int_literal OFFSET int_literal { $$ = new LimitDescription($2->ival, $4->ival); delete $2; delete $4; }
-	|	OFFSET int_literal { $$ = new LimitDescription(kNoLimit, $2->ival); delete $2; }
-	|	LIMIT ALL { $$ = nullptr; }
-	|	LIMIT NULL { $$ = nullptr;  }
-	|	LIMIT ALL OFFSET int_literal { $$ = new LimitDescription(kNoLimit, $4->ival); delete $4; }
-	|	LIMIT NULL OFFSET int_literal { $$ = new LimitDescription(kNoLimit, $4->ival); delete $4; }
+		LIMIT expr { $$ = new LimitDescription($2, Expr::makeNullLiteral()); }
+	|	LIMIT expr OFFSET expr { $$ = new LimitDescription($2, $4); }
```

Comment:
why use NULL instead of nullptr? I do not know whether `LIMIT 5 OFFSET NULL` has a defined semantic, but should it really be indifferentiable from `LIMIT 5`? For MySQL, at least, `LIMIT 5 OFFSET NULL` is no valid input, but `LIMIT 5` is, so in order to be able to tell those two apart I think

`LIMIT 5 OFFSET NULL` -> $$ = new LimitDescription($2, Expr::makeNullLiteral()); 
`LIMIT 5` -> $$ = new LimitDescription($2, nullptr); 
`LIMIT ALL OFFSET expr` -> $$ = new LimitDescription(nullptr, $4);

Reply:
Yup makes sense. I originally used nullptr actually, but changed it to Expr::Null, to have consistency with things like `LIMIT 5 OFFSET NULL`. But you're right, we actually want those to be different. I updated it now.